operator/standalone/create.go

```diff
@@ -0,0 +1,236 @@
+
+	instance   *v1alpha1.PostgresqlStandalone
+	config     *v1alpha1.PostgresqlStandaloneOperatorConfig
+	helmValues HelmValues
+	helmChart  *v1alpha1.ChartMeta
+}
+
+// NewCreateStandalonePipeline creates a new pipeline with the required dependencies.
+func NewCreateStandalonePipeline(client client.Client, instance *v1alpha1.PostgresqlStandalone, operatorNamespace string) *CreateStandalonePipeline {
+	return &CreateStandalonePipeline{
+		instance:          instance,
+		client:            client,
+		operatorNamespace: operatorNamespace,
+	}
+}
+
+// RunPipeline executes the pipeline with configured business logic steps.
+// This should only be executed once per pipeline as it stores intermediate results in the struct.
+func (p *CreateStandalonePipeline) RunPipeline(ctx context.Context) error {
+	return pipeline.NewPipeline().
+		WithSteps(
+			pipeline.NewStepFromFunc("fetch operator config", p.FetchOperatorConfig),
+
+			pipeline.NewPipeline().WithNestedSteps("compile helm values",
+				pipeline.NewStepFromFunc("read template values", p.UseTemplateValues),
+				pipeline.NewStepFromFunc("override template values", p.OverrideTemplateValues),
+				pipeline.NewStepFromFunc("apply values from instance", p.ApplyValuesFromInstance),
+			),
+
+			pipeline.NewPipeline().WithNestedSteps("deploy resources",
+				pipeline.NewStepFromFunc("ensure deployment namespace", p.EnsureDeploymentNamespace),
+				pipeline.NewStepFromFunc("ensure credentials secret", p.EnsureCredentialsSecret),
+				pipeline.NewStepFromFunc("ensure helmrelease exists", p.EnsureHelmRelease),
+			),
+		).
+		RunWithContext(ctx).Err()
+}
+
+// FetchOperatorConfig fetches a matching v1alpha1.PostgresqlStandaloneOperatorConfig from the OperatorNamespace.
+// The Major version specified in v1alpha1.PostgresqlStandalone is used to filter the correct config by the v1alpha1.PostgresqlMajorVersionLabelKey label.
+// If there is none or multiple found, it returns an error.
+func (p *CreateStandalonePipeline) FetchOperatorConfig(ctx context.Context) error {
+	list := &v1alpha1.PostgresqlStandaloneOperatorConfigList{}
+	labels := client.MatchingLabels{
+		v1alpha1.PostgresqlMajorVersionLabelKey: p.instance.Spec.Parameters.MajorVersion.String(),
+	}
+	ns := client.InNamespace(p.operatorNamespace)
+	err := p.client.List(ctx, list, labels, ns)
+	if err != nil {
+		return err
+	}
+	if len(list.Items) == 0 {
+		return fmt.Errorf("no %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	if len(list.Items) > 1 {
+		return fmt.Errorf("multiple versions of %s found with label '%s' in namespace '%s'", v1alpha1.PostgresqlStandaloneOperatorConfigKind, labels, ns)
+	}
+	p.config = &list.Items[0]
+	return nil
+}
+
+// UseTemplateValues copies the Helm values and Chart metadata from the v1alpha1.PostgresqlStandaloneOperatorConfig spec as the starting parameters.
+//
+// This step assumes that the config has been fetched first via FetchOperatorConfig.
```

Comment:
Maybe add a check for this in the function? Could lead to nil pointers otherwise.

Reply:
So it's not possible that a user error could result in a nil config here? If that's the case, then it's fine by me.